src/main/java/upb/dice/twitter/TweetSearchApplication.java

```diff
@@ -1,156 +1,75 @@
-        long key_id_1 = keyParse(key);
-        Query query1 = new Query(key);
-        if (key_id_1 != 1) {
-            query1.sinceId(key_id_1);
-        }
-        getTweet(query1);
-    }
+                //create available options
+                Option option1 = Option.builder("k").hasArg(true).numberOfArgs(1).desc("Keyword Based Search").longOpt("key").argName("Keyword").build();
+                Option option2 = Option.builder("l").longOpt("loc").hasArg(true).numberOfArgs(3).desc("Location Based Search").argName("Latitude><Longitude><Radius In Kilometers").valueSeparator(' ').build();
+                Option option3 = Option.builder("h").longOpt("Help").hasArg(false).desc("Help Menu").build();
 
-    //Location based Tweet Search
-    private void queryBuild(double latitude, double longitude, double radius) throws TwitterException, IOException {
-        GeoQuery geoQuery = new GeoQuery(new GeoLocation(latitude, longitude));
-        Query query = new Query();
-        query.setGeoCode(geoQuery.getLocation(), radius, Query.KILOMETERS);
-        lat = latitude;
-        lon = longitude;
+                //prepare the options
+                Options options = new Options();
+                options.addOption(option1).addOption(option2).addOption(option3);
 
-//       query.until("2019-02-31").since("2019-02-01");   //Specific time period
-        long key_id_1 = locationParse(new GeoLocation(latitude, longitude));
-        if (key_id_1 != 1) {
-            query.sinceId(key_id_1);
-        }
-        getTweet(query);
-    }
+                CommandLine cmd = null;
+                try {
+                    cmd = parser.parse(options, args);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
 
+                //if there is a value for keybased, get it
+                assert cmd != null;
+                if (cmd.hasOption("k")) {
+                    String[] searchArgs = cmd.getOptionValues("k");
 
-    //Extract the tweets based on the query built
-    private void getTweet(Query query) throws TwitterException, IOException {
-        QueryResult queryResult;
-        System.out.println("Tweets search Start");
-        FileWriter writer = new FileWriter("Tweets.txt", true);
-        int counter = 0;
-        do {
-            queryResult = twitter.search(query);
-            List<Status> queryTweets = queryResult.getTweets();
-            long maxId;
-            if (counter == 0) {
-                maxId = queryResult.getMaxId();
-                //Location based
-                if (query.getQuery() == null) {
-                    try (BufferedWriter out = new BufferedWriter(new FileWriter("Latitude_Longitude_MaxID.txt", true))) {
-                        out.append(String.valueOf(lat)).append(":").append(String.valueOf(lon)).append(":").append(String.valueOf(maxId));
-                        out.append('\n');
+                    String key = searchArgs[0];
```

Comment:
The command line parsing shouldn't take place within the timer task. It is sufficient if it is done once.